### ISSUE TITLE
Error occurred in poll cycle: uninitialized constant PumaAgent::Agent::Faraday

### ISSUE THREAD

**Issue by likethesky** [unboxed/newrelic_puma_plugin#1]
Any idea why I'm seeing this error when running _newrelic_puma_agent_ ?

**Comment by pawel2105** [comment]
@grantspeelman any idea?

**Comment by grantspeelman** [comment]
@likethesky Did you run "bundle install" ?

**Comment by leocassarani** [comment]
Could it be that Faraday is not required at the top of the file?

**Comment by likethesky** [comment]
Yes indeed, I did a bundle install. Wouldn't run otherwise, of course. 

Faraday bundled fine.

~ Brad

**Comment by likethesky** [comment]
So... I don't know why I'm seeing that.  I am using JRuby, which I assume a lot of folks using Puma are.

Here's a full gist of the output I'm seeing (over two one-minute periods): https://gist.github.com/likethesky/19aa67ac2b1a54ab1d04

I also get a 'WARN: Component with name "myserver.example.com" and guid "com.ubxd.newrelic_puma" had no metrics' along with the 'ERROR: Error occurred in poll cycle: uninitialized constant PumaAgent::Agent::Faraday' ...

I assume that the ERROR is more pertinent, but perhaps the WARN gives you a hint as to what's happening? (I also included a couple of outputs in my gist with verbose = 1 set as well.)

**Comment by svenagnew** [comment]
@likethesky - Can you tell me which version of JRuby you are using exactly? I'm going to try and reproduce the issue on my machine and see if I can shed some light on this issue.

**Comment by likethesky** [comment]
jruby 1.7.4 (1.9.3p392) 2013-05-16 2390d3b on Java HotSpot(TM) Client VM 1.6.0_51-b11-456-11M4508 [darwin-i386]

**Comment by likethesky** [comment]
Any luck reproducing, @svenagnew ?

**Comment by grantspeelman** [comment]
@likethesky I have release a new version that has the require of  Faraday like @leocassarani suggested

https://github.com/unboxed/newrelic_puma_plugin/tags

Will investigate the Warnings further
